core/src/main/kotlin/io/customer/sdk/core/util/ScopeProvider.kt

```diff
@@ -5,9 +5,12 @@ import kotlinx.coroutines.SupervisorJob
 
 interface ScopeProvider {
     val eventBusScope: CoroutineScope
+    val lifecycleListenerScope: CoroutineScope
 }
 
 class SdkScopeProvider(private val dispatchers: DispatchersProvider) : ScopeProvider {
     override val eventBusScope: CoroutineScope
         get() = CoroutineScope(dispatchers.default + SupervisorJob())
+    override val lifecycleListenerScope: CoroutineScope
+        get() = CoroutineScope(dispatchers.default + SupervisorJob())
```

Comment:
Although we are accessing these coroutines only once, I still think we should update them as follows to avoid recreating them every time.

```
override val eventBusScope: CoroutineScope = CoroutineScope(dispatchers.default + SupervisorJob())
override val lifecycleListenerScope: CoroutineScope = CoroutineScope(dispatchers.default + SupervisorJob())
```